File: src/main/java/com/epicsquid/whoosh/gui/ElementPointListBox.java
package com.epicsquid.whoosh.gui;

import cofh.core.client.gui.element.ElementListBox;
import cofh.core.client.gui.element.listbox.IListBoxElement;
import com.epicsquid.whoosh.utils.TransporterPoint;

import java.util.function.Consumer;

public class ElementPointListBox extends ElementListBox {

	private final Consumer<TransporterPoint> onChange;

	public ElementPointListBox(TransportScreen screen, int x, int y, int width, int height, Consumer<TransporterPoint> onChange) {
		super(screen, x, y, width, height);
		this.onChange = onChange;
	}

	@Override
	protected void onSelectionChanged(int newIndex, IListBoxElement newElement) {
		super.onSelectionChanged(newIndex, newElement);
		if (newElement instanceof ListBoxPointElement listBoxPointElement) {
			onChange.accept(listBoxPointElement.getValue());
		}
	}

	@Override
	public void add(IListBoxElement element) {
		super.add(element);
		if (_elements.size() == 1) {
			setSelectedIndex(0);
		}
	}
}

File: src/main/java/com/epicsquid/whoosh/gui/ElementPointListBox.kt
package com.epicsquid.whoosh.gui

import cofh.core.client.gui.element.ElementListBox
import cofh.core.client.gui.element.listbox.IListBoxElement
import com.epicsquid.whoosh.utils.TransporterPoint
import java.util.function.Consumer

class ElementPointListBox(
    screen: TransportScreen?,
    x: Int,
    y: Int,
    width: Int,
    height: Int,
    private val onChange: Consumer<TransporterPoint>
) : ElementListBox(screen, x, y, width, height) {
    override fun onSelectionChanged(newIndex: Int, newElement: IListBoxElement) {
        super.onSelectionChanged(newIndex, newElement)
        if (newElement is ListBoxPointElement) {
            onChange.accept(newElement.value)
        }
    }

    override fun add(element: IListBoxElement) {
        super.add(element)
        if (_elements.size == 1) {
            selectedIndex = 0
        }
    }
}
